### PR TITLE
initialize account balances in genesis

### LINKED ISSUE #1173: Confirm creating accounts using Polkadot.js
- Make the call and getBalance (resolve any issues with this)

Acceptance Criteria
- Polkadot.js API getBalance to return an address using Gossamer

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes

- Added entries in genesis-raw.json to initialize test accounts: `Alice, Bob, Charlie, Dave, Eve, Ferdie, George, Heather, Ian` 
- Added entries `balances:balances` to initialize human readable genesis with account balances.
- Updated `build-spec` action to use balances:balances entry to build raw genesis files.

## Tests

```
go test ./lib/genesis/...
```
Test with polkadot.js api example script:
```
// Import the API
const { ApiPromise, WsProvider } = require('@polkadot/api');

// Known account we want to use (available on dev chain, with funds)
const Alice = '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY';

async function main () {
        const provider = new WsProvider('ws://127.0.0.1:8546');

        // Create an await for the API
        const api = await ApiPromise.create({provider});

        freeBal = await api.query.system.account(Alice);
        console.log(`Alice free bal ${JSON.stringify(freeBal)}`);
}

main().catch(console.error);
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- closes #1173

### CODE REVIEW

File: lib/genesis/helpers.go [hunk 1]
```diff
@@ -189,7 +190,14 @@ func buildRawMap(m map[string]map[string]interface{}) (map[string]string, error)
 		kv.key = append(kv.key, k)
 		buildRawMapInterface(v, kv)
 
-		key, err := formatKey(kv.key)
+		if reflect.DeepEqual([]string{"balances", "balances"}, kv.key) {
```
Comment (noot) [inline lib/genesis/helpers.go:193]: I was wondering why it uses balances:balances here, but `System Account` in `buildBalances`?

**Review by noot — APPROVED** [review]
looks good to me! please run `go mod tidy`

also, I tried out the script, ran the node with `./bin/gossamer --key alice --rpc --ws` and got this:
```
$ node balanceTest.js 
2020-12-14 12:19:23        RPC-CORE: methods(): RpcMethods:: -32000: rpc: can't find service "rpc.Methods": null
Alice free bal {"nonce":0,"refcount":0,"data":{"free":68719476736,"reserved":0,"miscFrozen":0,"feeFrozen":0}}
```

what is the first error? I guess it's unrelated since I was able to see the balance?

**Review by RyRy79261 — APPROVED** [review]

File: lib/genesis/helpers.go (hunk 1 above)
Comment (edwardmack) [reply lib/genesis/helpers.go:193]: This was based on an older substrate genesis file, newer genesis files have `palletBalances:balances`.  I'd like to determine a "standard" way of building these, so I'm open changing this.  Let me know.

**Comment by edwardmack** [comment]
@noot Re rpc_methods error, when polkadot.js-api connects it calls `rpc_methods` to learn what methods are available, however our default configuration does not enable the "rpc" module.  We should probably consider changing this.

**Comment by noot** [comment]
@edwardmack could you update the PR to enable the `rpc` module by default?

File: lib/genesis/helpers.go (hunk 1 above)
Comment (noot) [reply lib/genesis/helpers.go:193]: i see, could you update it to the newer version for now? i think as long as we keep it up to date with substrate's genesis it's good

File: dot/state/storage_notify.go [hunk 2]
```diff
@@ -74,3 +77,16 @@ func (s *StorageState) notifyChanged(change *KeyValue) {
 		}(ch)
 	}
 }
+
+func (s *StorageState)Subscribe(prefixes []byte) {
+	fmt.Printf("SUBSCRIBE CALLED db %v pre %x\n", s.db, prefixes)
```
Comment (arijitAD) [inline dot/state/storage_notify.go:82]: Can we replace this with logger debug/trace?

File: dot/rpc/modules/api.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ type StorageAPI interface {
 	RegisterStorageChangeChannel(ch chan<- *state.KeyValue) (byte, error)
 	UnregisterStorageChangeChannel(id byte)
 	GetStateRootFromBlock(bhash *common.Hash) (*common.Hash, error)
+	Subscribe(prefixes []byte)
```
Comment (arijitAD) [inline dot/rpc/modules/api.go:22]: Are we planning to subscribe to a single prefix or multiple?

File: lib/genesis/helpers.go [hunk 4]
```diff
@@ -225,30 +233,32 @@ func buildRawArrayInterface(a []interface{}, kv *keyValue) {
...
 		kb := []byte(`:code`)
 		return common.BytesToHex(kb), nil
 	default:
-		prefix, err := common.Twox128Hash([]byte(key[0]))
+		prefix, err := common.Twox128Hash([]byte(kv.key[0]))
 		if err != nil {
 			return "", err
 		}
-		keydata, err := common.Twox128Hash([]byte(key[1]))
+		keydata, err := common.Twox128Hash([]byte(kv.key[1]))
```
Comment (arijitAD) [inline lib/genesis/helpers.go:261]: Should we have a length check? It might segfault if length is 0 or 1.

File: dot/state/storage_notify.go (hunk 2 above)
Comment (edwardmack) [reply dot/state/storage_notify.go]: Yes, I will (this was from a false commit from another branch)

File: dot/rpc/modules/api.go (hunk 3 above)
Comment (edwardmack) [reply dot/rpc/modules/api.go]: Potentially multiple.

File: lib/genesis/helpers.go (hunk 4 above)
Comment (edwardmack) [reply lib/genesis/helpers.go:265]: Yes, good idea.

**Pull request merged**